### PR TITLE
Lemma Bid Adapter: accepting the floor to use the getFloor function

### PR DESCRIPTION
**Pull request by lemmadev**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@lemmadev can you please add test coverage for some of these changes. Looks like you are only removing test coverage

**Comment by lemmadev** [comment]
Made required changes. Please review it.

File: modules/lemmaBidAdapter.js [hunk 1]
```diff
@@ -368,11 +376,9 @@ function _getImpressionObject(bid) {
...
     bidfloorcur: params.currency ? params.currency : DEFAULT_CURRENCY
   };
-
-  if (params.bidFloor) {
-    impression.bidfloor = params.bidFloor;
-  }
-
+  
+  _setFloor(impression, bid);
+
```
Comment (FilipStamenkovic) [inline modules/lemmaBidAdapter.js:381]: You have lint error here.  
Check output of circle ci, or just run `gulp lint` on your local to fix this error.

File: modules/lemmaBidAdapter.js [hunk 2]
```diff
@@ -62,18 +62,6 @@ export var spec = {
...
-    if (gdprConsent) {
-      syncurl += '&gdpr=' + (gdprConsent.gdprApplies ? 1 : 0);
-      syncurl += '&gdpr_consent=' + encodeURIComponent(gdprConsent.consentString || '');
-    }
-
-    // CCPA
-    if (uspConsent) {
-      syncurl += '&us_privacy=' + encodeURIComponent(uspConsent);
-    }
-
```
Comment (FilipStamenkovic) [inline modules/lemmaBidAdapter.js:76]: Why removing gdpr and ccpa support?  
If you really need to remove this, please do it in separate PR. This PR should be only about floor prices.

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]
Take a look at my comments.  
Also, please merge/rebase your branch with origin/master, your PR is 300+ commits behind master branch.  

Add unit tests for the changes you made. @ChrisHuie already mentioned this.

File: modules/lemmaBidAdapter.js (hunk 2 above)
Comment (lemmadev) [reply modules/lemmaBidAdapter.js]: Hi, 
Above changes already merged into master 
https://github.com/prebid/Prebid.js/pull/6006

File: modules/lemmaBidAdapter.js (hunk 1 above)
Comment (lemmadev) [reply modules/lemmaBidAdapter.js:381]: I have run gulp lint and there is not showing any error on the terminal.

Comment (FilipStamenkovic) [reply modules/lemmaBidAdapter.js:381]: It won't show you any error, it will automatically fix it.  
You have trailing spaces on lines 379 and 381, see circle ci output.

Comment (lemmadev) [reply modules/lemmaBidAdapter.js]: Remove trailing spaces on lines 379 and 381.

File: test/spec/modules/lemmaBidAdapter_spec.js [hunk 3]
```diff
@@ -161,6 +161,7 @@ describe('lemmaBidAdapter', function() {
         expect(data.site.publisher.id).to.equal(bidRequests[0].params.pubId.toString()); // publisher Id
         expect(data.imp[0].tagid).to.equal('1'); // tagid
         expect(data.imp[0].bidfloorcur).to.equal(bidRequests[0].params.currency);
+        expect(data.imp[0].bidfloor).to.equal(bidRequests[0].params.floor);
```
Comment (FilipStamenkovic) [inline test/spec/modules/lemmaBidAdapter_spec.js:164]: You should add tests for the changes you made.  
This added line is not testing your changes, here you are only comparing `undefined` with `undefined`.  

Run `gulp test-coverage` and `gulp view-coverage` to see your coverage.

File: test/spec/modules/lemmaBidAdapter_spec.js [hunk 4]
```diff
@@ -192,6 +193,7 @@ describe('lemmaBidAdapter', function() {
         expect(data.site.publisher.id).to.equal(bidRequests[0].params.pubId.toString()); // publisher Id
         expect(data.imp[0].tagid).to.equal(undefined); // tagid
         expect(data.imp[0].bidfloorcur).to.equal(bidRequests[0].params.currency);
+        expect(data.imp[0].bidfloor).to.equal(bidRequests[0].params.floor);
```
Comment (FilipStamenkovic) [inline test/spec/modules/lemmaBidAdapter_spec.js:196]: Here you are also comparing `undefined` with `undefined`.

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]
Please add unit tests for the changes.

File: test/spec/modules/lemmaBidAdapter_spec.js (hunk 3 above)
Comment (lemmadev) [reply test/spec/modules/lemmaBidAdapter_spec.js:166]: I have added test coverage for getFloor function changes.

File: test/spec/modules/lemmaBidAdapter_spec.js (hunk 4 above)
Comment (lemmadev) [reply test/spec/modules/lemmaBidAdapter_spec.js:198]: Added test coverage for getFloor function changes.

File: test/spec/modules/lemmaBidAdapter_spec.js [hunk 5]
```diff
@@ -309,6 +313,77 @@ describe('lemmaBidAdapter', function() {
...
+        };
+
+        beforeEach(() => {
+          floorModuleTestData = {
+            'banner': {
+              'currency': 'USD',
+              'floor': 1.50
+            },
+            'video': {
+              'currency': 'USD',
```
Comment (FilipStamenkovic) [inline test/spec/modules/lemmaBidAdapter_spec.js:336]: For currency please use `AUD`, in all your tests you have `params.currency: 'AUD'`, so following if condition is never true:

```
if (typeof floorInfo === 'object' && floorInfo.currency === impObj.bidfloorcur && !isNaN(parseInt(floorInfo.floor)))
```

Because `fllorInfo.currency` in your tests is always `USD` while `impOb.jbidfloorcur` is always `AUD`

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]

File: test/spec/modules/lemmaBidAdapter_spec.js (hunk 5 above)
Comment (lemmadev) [reply test/spec/modules/lemmaBidAdapter_spec.js]: Made suggested changes.

**Review by FilipStamenkovic — APPROVED** [review]
